feat: ✨ add section about GitHub teams and how we use them (#372)
seedcase-project/team: how-we-work/github.qmd

one-time password (TOTP) apps, good ones are [Google
Authenticator](https://support.google.com/accounts/answer/1066447?hl=en&ref_topic=2954345&sjid=1964065568409601298-EU)
or [Microsoft
Authenticator](https://www.microsoft.com/en-ww/security/mobile-authenticator-app).

<!-- ## Using GitHub Projects for managing tasks -->

## Using GitHub Teams

We use GitHub
[teams](https://docs.github.com/en/organizations/organizing-members-into-teams/about-teams)
to manage permissions on repositories, effectively use notifications, and
communicate ownership of repositories and content. We have two higher-level
categories of teams that have write access to their assigned repositories:

- A "product" category for things that (we believe) give direct and clear value
  to others.
- A "platform" category for things that are mainly about improving our own work
  on developing the products.

Within those two categories, we have two or three subcategories for teams:

- "tools" for software packages or other code-based resources.
- "docs" for documentation-only websites.
- "templates" for templates for different types of repositories (e.g. Python

>   package).

Reviewer: ```suggestion
  packages).
```

  them.

- Use the read-only "everyone" team when we want to get reviews on pull requests
  that impact everyone (e.g. design decisions).

- Each repository is owned by one team with write access (not admin or maintain
  permissions) who is primarily responsible for doing work in the repository.
  That team is included in the `CODEOWNERS` file (though a `CODEOWNERS` file
  could have multiple teams in it) to clearly communicate who owns the
  repository and has governance over it.

- All pull requests must have a review from someone of the team that owns the
  repository before it can be merged (enforced with
  [rulesets](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets)).
  Codeowners are automatically requested for reviews from pull requests in that
  repository, which means people within the team will get a notification from
  that request.

- If someone is on a team and gets assigned for review, they are expected to
  review those pull requests.

- People are only added to a team when there is work to be done on a repository
  in that team for a given iteration. This limits the number of notifications

>   that everyone in Seedcase will get, so that only those relevant people are

Reviewer: ```suggestion
  that everyone in Seedcase gets, so that only those relevant people are
```

  notified.

- The admin team/organization owners are responsible for adding or removing

>   people from teams as needed, based on an iteration's aims.

Reviewer: Does that mean I'm responsible for this too?

Author: Yea, though really only if I get hit by a bus or something 😛 🚌

> - A member is on a team if they are assigned work on the repository.

Reviewer: ```suggestion
- A member is on a team if they are assigned work on the repository for the current iteration.
```
Maybe?

- Each higher-level grouping of teams ("product" or "platform") has their own
  project board. Multiple teams can work on the same project board, as long as
  the work is under the same grouping (e.g. "platform-tools" and "platform-docs"

>   can share a project board).

Reviewer: ```suggestion
  are both on the platform project board).
```
Maybe?
